samples/monster_generated.rs

```diff
@@ -1,148 +1,163 @@
 // automatically generated by the FlatBuffers compiler, do not modify
 
 
+#![allow(unused_imports, dead_code)]
+
+use std::mem;
+use std::cmp::Ordering;
+
+extern crate flatbuffers;
+use self::flatbuffers::EndianScalar;
+
+#[allow(unused_imports, dead_code)]
 pub mod my_game {
-  #![allow(dead_code)]
-  #![allow(unused_imports)]
 
   use std::mem;
-  use std::marker::PhantomData;
   use std::cmp::Ordering;
 
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
+#[allow(unused_imports, dead_code)]
 pub mod sample {
-  #![allow(dead_code)]
-  #![allow(unused_imports)]
 
   use std::mem;
-  use std::marker::PhantomData;
   use std::cmp::Ordering;
 
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
 
-#[allow(non_camel_case_types)]
-#[repr(i8)]
-#[derive(Clone, Copy, PartialEq, Debug)]
-pub enum Color {
-  Red = 0,
-  Green = 1,
-  Blue = 2
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Color(pub i8);
```

Comment:
Inner variable should not be public. We don't want to allow users mutate it directly, because this is enum, not an integer.

Reply:
I think it should be public so you can construct an enum that's "invalid" to you but may be valid for some other service, due to decoupled releases. E.g. a translation and routing server may want to copy json to flatbuffers (and the enum is a json int for some reason) and pass along an unrecognized enum.

Granted, this could be achieved with a `new` and `get` method (omitting `set`) but I'd rather not be too prescriptive.

Reply:
The problem here is that user can do
```rust
let mut c = Color::Red;
c.0 += 1;
```

Reply:
sure, but the dumb thing will still be possible

```rust
c = Color::new(Color::Red.get() + 1);
```

Reply:
Yes, but that "doesn't make sense" thing would take more effort to write.

Welp, maybe `pub` field isn't that bad, and it's just me not liking changing enums through addition instead of matching.